pub_key.go

```diff
@@ -15,7 +18,77 @@ import (
 // An address is a []byte, but hex-encoded even in JSON.
 // []byte leaves us the option to change the address length.
 // Use an alias so Unmarshal methods (with ptr receivers) are available too.
-type Address = cmn.HexBytes
+type Address struct {
+	cmn.HexBytes
+}
+
+func (a *Address) UnmarshalJSON(data []byte) error {
+	readable, deserialized, err := bech32cosmos.Decode(string(data))
+	if err != nil {
+		return err
+	}
+	if strings.ToUpper(readable) != "CSMSADDR" {
+		return fmt.Errorf("%s is not CSMSADDR the Cosmos Address identifier", readable)
+	}
+
+	err = cdc.UnmarshalBinary(deserialized, a)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (a *Address) MarshalJSON() ([]byte, error) {
```

Comment:
can't this just call String()?

Reply:
So I wanted to preserve the same API as `HexBytes` offered and I need an override here because `HexBytes` implements `MarshalJSON`/`UnmarshalJSON`